### ISSUE TITLE
Multiple positional arguments are greedy (order is not maintained)

### ISSUE THREAD

**Issue by kelvinhammond** [CLIUtils/CLI11#295]
Positional arguments 

```
std::vector<std::string> sources;
std::string dest;
sub->add_option("src", sources, "Source files");
sub->add_option("dest", dest, "Destination folder or directory");
```
This is the output from `./cmd 1 2 3
```
src: 1 2 3
dest: 
```

The first `src` argument consumes the last argument which should be `dest`.

**Comment by WenbinHou** [comment]
I'm facing the same issue. Any ideas please?

Set `src` and `dest` as required() does not work.

**Comment by phlptp** [comment]
I added a solution in PR #306.   The way it would work is 
```cpp
std::vector<std::string> sources;
std::string dest;
sub->add_option("src", sources, "Source files");
sub->add_option("dest", dest, "Destination folder or directory")->required();
sub->positionals_at_end();
```

The other option that I think would work on the existing code is to apply validators on the positionals  and the 'validate_positionals()'  flag,  then if source files have to be existing files and dest is a folder they can have different validators and the positional validation will not load them if they don't validate and move on to the next one.    That isn't a general solution but if you are actually looking at files that should work.

**Comment by henryiii** [comment]
Fixed in #306 and will be released in 1.9.
